src/Http/Http/perf/SendFileBenchmark.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.IO.Pipelines;
+using System.Threading;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+
+namespace Microsoft.AspNetCore.Http.Extensions
+{
+    public class SendFileBenchmark
```

Comment:
@davidfowl Is this the benchmark you've expected?

I think I've done something wrong! I expected to see better result for pipeline. I don't know if it's because of the `MemoryStream` or the way I'm creating `PipeWriter`.


| Method | FileSize |         Mean |        Error |       StdDev |      Op/s |    Gen 0 |    Gen 1 |    Gen 2 |    Allocated |
|------- |--------- |-------------:|-------------:|-------------:|----------:|---------:|---------:|---------:|-------------:|
|   Pipe |      512 |     334.5 us |      1.98 us |      1.76 us | 2,989.614 |   0.4883 |        - |        - |     21.82 KB |
| Stream |      512 |     330.2 us |      2.04 us |      1.81 us | 3,028.014 |   0.4883 |        - |        - |     17.59 KB |
|   Pipe |     1024 |     335.9 us |      4.53 us |      4.65 us | 2,976.832 |   0.4883 |        - |        - |     22.33 KB |
| Stream |     1024 |     330.7 us |      1.77 us |      1.65 us | 3,024.013 |   0.4883 |        - |        - |      18.1 KB |
|   Pipe |     2048 |     339.3 us |      2.65 us |      2.35 us | 2,947.565 |   0.4883 |        - |        - |     23.33 KB |
| Stream |     2048 |     340.7 us |      6.75 us |      8.54 us | 2,934.737 |   0.4883 |        - |        - |      19.1 KB |
|   Pipe |     4096 |     338.5 us |      2.50 us |      2.34 us | 2,954.512 |   0.9766 |        - |        - |     25.33 KB |
| Stream |     4096 |     333.6 us |      2.68 us |      2.38 us | 2,997.450 |   0.4883 |        - |        - |      21.1 KB |
|   Pipe |     8192 |     352.4 us |      6.92 us |      8.50 us | 2,837.805 |   0.9766 |        - |        - |     33.35 KB |
| Stream |     8192 |     337.5 us |      3.15 us |      2.95 us | 2,963.051 |   0.9766 |        - |        - |     29.13 KB |
|   Pipe |    16384 |     346.7 us |      2.02 us |      1.79 us | 2,884.137 |   1.4648 |        - |        - |     49.38 KB |
| Stream |    16384 |     349.6 us |      6.27 us |      7.23 us | 2,860.689 |   1.4648 |        - |        - |     45.15 KB |
|   Pipe |    32768 |     431.1 us |      4.59 us |      4.30 us | 2,319.587 |   2.9297 |        - |        - |     81.54 KB |
| Stream |    32768 |     424.6 us |      3.77 us |      3.53 us | 2,354.890 |   2.4414 |        - |        - |     77.31 KB |
|   Pipe |    65536 |     595.9 us |      6.74 us |      5.63 us | 1,678.087 |   4.8828 |        - |        - |    145.84 KB |
| Stream |    65536 |     580.5 us |      3.91 us |      3.47 us | 1,722.751 |   4.8828 |        - |        - |    141.62 KB |
|   Pipe |   131072 |   1,175.6 us |     21.97 us |     20.55 us |   850.614 |  11.7188 |   7.8125 |   7.8125 |    274.49 KB |
| Stream |   131072 |   1,140.3 us |     18.33 us |     17.14 us |   876.942 |  11.7188 |   7.8125 |   7.8125 |    270.25 KB |
|   Pipe |   262144 |   2,244.1 us |     42.86 us |     40.09 us |   445.613 |  23.4375 |  19.5313 |  19.5313 |    531.72 KB |
| Stream |   262144 |   2,130.0 us |     31.98 us |     29.91 us |   469.484 |  23.4375 |  19.5313 |  19.5313 |     527.5 KB |
|   Pipe |   524288 |   4,404.4 us |     86.47 us |    102.94 us |   227.044 |  46.8750 |  46.8750 |  46.8750 |   1046.21 KB |
| Stream |   524288 |   4,166.9 us |     70.05 us |     62.09 us |   239.984 |  46.8750 |  46.8750 |  46.8750 |   1041.92 KB |
|   Pipe |  1048576 |   8,786.0 us |    171.90 us |    176.52 us |   113.817 |  78.1250 |  78.1250 |  78.1250 |   2074.73 KB |
| Stream |  1048576 |   8,268.6 us |    156.81 us |    167.79 us |   120.939 |  78.1250 |  78.1250 |  78.1250 |   2070.62 KB |
|   Pipe |  2097152 |  17,526.6 us |    202.30 us |    157.94 us |    57.056 | 125.0000 | 125.0000 | 125.0000 |   4131.87 KB |
| Stream |  2097152 |  16,579.7 us |    325.08 us |    361.33 us |    60.315 | 156.2500 | 156.2500 | 156.2500 |    4128.1 KB |
|   Pipe |  4194304 |  36,566.0 us |    726.26 us |    969.54 us |    27.348 | 214.2857 | 214.2857 | 214.2857 |   8245.98 KB |
| Stream |  4194304 |  33,833.3 us |    381.19 us |    356.56 us |    29.557 | 200.0000 | 200.0000 | 200.0000 |    8242.2 KB |
|   Pipe |  8388608 |  72,931.2 us |  1,025.62 us |    959.37 us |    13.712 | 285.7143 | 285.7143 | 285.7143 |  16473.51 KB |
| Stream |  8388608 |  68,822.0 us |    838.08 us |    783.94 us |    14.530 | 250.0000 | 250.0000 | 250.0000 |  16470.18 KB |
|   Pipe | 16777216 | 148,980.0 us |  2,944.87 us |  3,273.22 us |     6.712 | 250.0000 | 250.0000 | 250.0000 |  32929.47 KB |
| Stream | 16777216 | 137,529.4 us |  2,183.99 us |  2,042.90 us |     7.271 | 250.0000 | 250.0000 | 250.0000 |  32925.43 KB |
|   Pipe | 33554432 | 290,936.3 us |  2,184.31 us |  2,043.20 us |     3.437 | 500.0000 | 500.0000 | 500.0000 |   65842.2 KB |
| Stream | 33554432 | 274,720.0 us |  1,797.73 us |  1,501.19 us |     3.640 |        - |        - |        - |  65837.89 KB |
|   Pipe | 67108864 | 592,565.0 us | 11,700.62 us | 15,214.11 us |     1.688 |        - |        - |        - | 131665.52 KB |
| Stream | 67108864 | 549,978.7 us |  4,076.52 us |  3,813.18 us |     1.818 |        - |        - |        - | 131661.42 KB |